src/main/java/com/wanderersoftherift/wotr/loot/functions/RollGearFunction.java

```diff
@@ -56,23 +59,28 @@ public LootItemFunctionType<? extends LootItemConditionalFunction> getType() {
 
     @Override
     protected ItemStack run(ItemStack itemStack, LootContext lootContext) {
-        return generateItemStack(itemStack, lootContext.getRandom());
+        return generateItemStack(itemStack, lootContext.getLevel(), lootContext.getRandom());
     }
 
-    private @NotNull ItemStack generateItemStack(ItemStack itemStack, RandomSource random) {
+    private @NotNull ItemStack generateItemStack(ItemStack itemStack, ServerLevel serverLevel, RandomSource random) {
 
         itemStack = ItemTagUtil.getRandomItemStackFromTag(itemStack, tagLocation, random);
         // todo add rift tier to ModDataComponentType for gear should it have its own place in implicits or does it need
         // to be its own ModDataComponentType?
-        // todo ask Pat if we want to roll implicit gear here or dot it JIT when you put it in your MH it may be better
-        // UX to do it here
         itemStack.set(ModDataComponentType.GEAR_SOCKETS, GearSockets.randomSockets(minSockets, maxSockets, random));
+        GearImplicits implicits = itemStack.get(ModDataComponentType.GEAR_IMPLICITS);
+        if (implicits != null) {
+            implicits.modifierInstances(itemStack, serverLevel);
+        }
+        RiftData riftData = RiftData.get(serverLevel);
```

Comment:
Should something be done to support rift completion rewards?

Reply:
Shit, forgot about those. Will adjust this to support that.